### PR TITLE
Registration  System

### PR DESCRIPTION
**Pull request by hanyulo**
Added features:

( will edit this later )

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -31,6 +32,36 @@ const store = configureStore(reduxState)
...
+  }
+  if(localStorage.getItem('token')) {
+    store.dispatch(authUserAction('verified token', ''))
+  }
+}
+const browserLocalStorage = (typeof localStorage === 'undefined') ? null : localStorage
+if(browserLocalStorage) {
+  // Check OAuth and store token
+  const { auth } = store.getState()
+  if( typeof localStorage !== 'undefined' ) {
```
Comment (garfieldduck) [inline src/index.js]: already checked in `if(browserLocalStorage) {`

Comment (hanyulo) [inline src/index.js]: Already refactored

File: src/index.js [hunk 2]
```diff
@@ -31,6 +32,18 @@ const store = configureStore(reduxState)
 
 const history = syncHistoryWithStore(browserHistory, store)
 
+// Check OAuth and store token
+const { auth } = store.getState()
+if(auth.authenticated && auth.authInfo && (auth.authType=== 'facebook' || auth.authType==='google')) {
+  const authInfo = auth.authInfo
+  setupToken(authInfo)
```
Comment (garfieldduck) [inline src/index.js]: setupToken(auth.authInfo)

File: src/index.js [hunk 3]
```diff
@@ -31,6 +32,18 @@ const store = configureStore(reduxState)
...
+  store.dispatch({ type: types.DELETE_AUTHINFO })
```
Comment (garfieldduck) [inline src/index.js]: could you make it as an action in `react-redux-registration`

File: server/server.js [hunk 4]
```diff
@@ -85,6 +90,32 @@ server.get('*', async function (req, res, next) {
 
   let location = createLocation(req.url)
 
+  // setup token to store state from cookies
+  if (req.query.login) {
+    let authType = req.query.login
```
Comment (garfieldduck) [inline server/server.js]: use const

File: server/server.js [hunk 5]
```diff
@@ -85,6 +90,32 @@ server.get('*', async function (req, res, next) {
...
+    let cookies = req.cookies
+    let auth_info_string = cookies.auth_info
+    let authInfoObj = authInfoStringToObj(auth_info_string)
```
Comment (garfieldduck) [inline server/server.js]: const authInfoObj = authInfoStringToObj(cookies.auth_info)

File: src/index.js (hunk 2 above)
Comment (hanyulo) [inline src/index.js]: Done.

File: src/index.js (hunk 3 above)
Comment (hanyulo) [inline src/index.js]: 👍

File: server/server.js (hunk 4 above)
Comment (hanyulo) [inline server/server.js]: Yes, sir!

**Pull request closed without merging**